src/atoms/TextInput/TextInput.web.js

```diff
@@ -326,7 +326,7 @@ const TextInput = ({
   );
 
   useEffect(() => {
-    if (value) {
+    if (value !== null && typeof value !== 'undefined') {
```

Comment:
I think having negation checks are hard to understand. Can we just do this?
`if(value=== '' || value)`